op-acceptor/cmd/main_test.go

```diff
@@ -123,63 +118,91 @@ func TestExitCodeBehavior(t *testing.T) {
 	}
 }
 
-func createMockGoMod(t *testing.T, testDir string) {
+// ensureBinaryExists builds the op-acceptor binary if it doesn't exist
+func ensureBinaryExists(t *testing.T, projectRoot, binaryPath string) {
+	// Build the binary if it doesn't exist
+	if !fileExists(binaryPath) {
+		t.Logf("Building op-acceptor binary...")
 
-	// Create go.mod file for the test module
-	goModPath := filepath.Join(testDir, "go.mod")
-	goModContent := `module github.com/test/test
+		// Create bin directory if needed
+		err := os.MkdirAll(filepath.Dir(binaryPath), 0755)
+		require.NoError(t, err, "Failed to create directory for binary")
 
-go 1.20
-`
+		// Build the binary
+		buildCmd := exec.Command("go", "build", "-o", binaryPath, filepath.Join(projectRoot, "cmd"))
+		var buildOutput bytes.Buffer
+		buildCmd.Stdout = &buildOutput
+		buildCmd.Stderr = &buildOutput
+
+		err = buildCmd.Run()
+		if err != nil {
+			t.Logf("Build output:\n%s", buildOutput.String())
+			t.Fatalf("Failed to build op-acceptor binary: %v", err)
+		}
 
-	require.NoError(t, os.WriteFile(goModPath, []byte(goModContent), 0644))
+		t.Logf("Successfully built binary at %s", binaryPath)
+	}
+
+	// Verify binary exists
+	require.FileExists(t, binaryPath, "op-acceptor binary not found")
```

Comment:
Maybe a tad stronger verification - check if file is executable

Reply:
Good thinking. I'll do this in a follow-up PR so this doesn't block downstream work. Thanks!